cocotb/triggers.py

```diff
@@ -199,7 +194,12 @@ def __init__(self, time_ps, units=None):
             :func:`~cocotb.utils.get_sim_steps`
         """
         GPITrigger.__init__(self)
-        self.sim_steps = get_sim_steps(time_ps, units)
+        self._time_ps = time_ps
+        self._units = units
+
+    @lazy_property
+    def sim_steps(self):
+        return get_sim_steps(self._time_ps, self._units)
```

Comment:
```suggestion
        # lazy so that we don't call into the simulator until we need to
        return get_sim_steps(self._time_ps, self._units)
```